File: backend/queute/src/main/java/tn/cloudnine/queute/model/roadmap/Step.java
package tn.cloudnine.queute.model.roadmap;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor

public class Step {
    @Id
    @GeneratedValue(strategy= GenerationType.IDENTITY)
    private Long id ;
    private int stepOrder;
    private boolean isStrict;
    private String stepName ;
    private String stepDescription ;
    private String requiredPapers ;

}

File: backend/template/src/app/models/Step.ts
    export class Step {
        id: number|null;
        stepOrder: number;
        isStrict: boolean;
        stepName: string;
        stepDescription: string;
        requiredPapers: string;
        
        constructor(init?: Partial<Step>) {
            this.id = init?.id ?? null;
            this.stepOrder = init?.stepOrder ?? 0;
            this.isStrict = init?.isStrict ?? false;
            this.stepName = init?.stepName ?? '';
            this.stepDescription = init?.stepDescription ?? '';
            this.requiredPapers = init?.requiredPapers ?? '';
        }
      
    }
